### DISCUSSION TITLE
Welcome to stm32f429 Discussions!

### DISCUSSION THREAD

**Discussion by MaJerle** [MaJerle/stm32f429#20] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JoergSH** [comment]
I wondering why the Libraries still not base on cube mx. Hardware configuration by cube and Your Libs depend on code generated by cube. That would be much easier for beginners and to use with other MCUs.

**Comment by MaJerle** [comment]
Reason is simple. CubeMX did not exist when these libs were written and since HAL exists, Idont see a point to make another HAL.

**Comment by ysolodkyyOTTO** [comment]
I've tried adopting your HAL Library 19- CPU LOAD monitor for an STM32F411E-discovery board. I keep getting 99% CPU loading, even when running no other code inside the main while loop. Any ideas what could be causing that? 
I followed your example as the starting point  https://stm32f4-discovery.net/2015/08/hal-library-19-cpu-load-monitor-for-stm32fxxx/

**Comment by SZPT18010137** [comment]
Your functional return type is interesting! Like:
uint8_t 
FunctionName(parameter ... )
When I first saw, I was attracted to your function name, then the return type

**Comment by EmbeddedLife615** [comment]
邮件已收到，我会尽快回复的！

**Comment by SZPT18010137** [comment]
Your functional return type is interesting! Like:
uint8_t 
FunctionName(parameter ... )
When I first saw, I was attracted to your function name, then the return type
